runtime/parachains/src/hrmp.rs

```diff
@@ -470,24 +532,35 @@ pub mod pallet {
 		///
 		/// If there are pending HRMP close channel requests, you can use this
 		/// function process all of those requests immediately.
-		#[pallet::weight(0)]
-		pub fn force_process_hrmp_close(origin: OriginFor<T>) -> DispatchResult {
+		///
+		/// Total number of closing channels must be provided as witness data of weighing.
+		#[pallet::weight(<T as Config>::WeightInfo::force_process_hrmp_close(*_channels))]
+		pub fn force_process_hrmp_close(origin: OriginFor<T>, _channels: u32) -> DispatchResult {
 			ensure_root(origin)?;
 			Self::process_hrmp_close_channel_requests();
 			Ok(())
 		}
 
-		/// This cancels a pending open channel request. It can be canceled be either of the sender
+		/// This cancels a pending open channel request. It can be canceled by either of the sender
 		/// or the recipient for that request. The origin must be either of those.
 		///
 		/// The cancellation happens immediately. It is not possible to cancel the request if it is
 		/// already accepted.
-		#[pallet::weight(0)]
+		///
+		/// Total number of open requests (i.e. `HrmpOpenChannelRequestsList`) must be provided as
+		/// witness data.
+		#[pallet::weight(<T as Config>::WeightInfo::hrmp_cancel_open_request(*open_requests))]
 		pub fn hrmp_cancel_open_request(
 			origin: OriginFor<T>,
 			channel_id: HrmpChannelId,
+			open_requests: u32,
 		) -> DispatchResult {
 			let origin = ensure_parachain(<T as Config>::Origin::from(origin))?;
+			ensure!(
+				<Self as Store>::HrmpOpenChannelRequestsList::decode_len().unwrap_or_default()
+					as u32 == open_requests,
```

Comment:
```suggestion
					as u32 <= open_requests,
```

Is this helpful since I assume open_requests is just needs to be an upper bound